### ISSUE TITLE
Share library loads very slowly

### ISSUE THREAD

**Issue by valleyofdawn** [IsraelHikingMap/Site#1294]
## Bug

Related to #777 

## What I expect to happen
I expect to see to top (most recent) shares within a few seconds of opening the "my shares" window

## What really happened
It takes ~30 seconds before the list begins to be populated

## What I did that caused the issue - step by step
Shared hundreds of recordings and trip plans over the years
Cleaned the cache
Opened the "my shares" window
Waited till I saw the first shares.

## My environment
Either browser or app

## Things I think the developers should know, images, links etc.
This is the regular delay, regardless of server slowdowns
It seems like a process is waiting for all the old shares to be retrieved before the first ones are shown
If we think sharing should be prioritized over saving, this should be optimized.

**Comment by HarelM** [comment]
Looking at the netwrok tab and the resulting response only the text of the shares is being sent, which is already optimized.
Can you please follow these steps:
1. press F12
2. Go to network tab
3. press clear ![image](https://user-images.githubusercontent.com/3269297/90749277-ee2f1680-e2db-11ea-939e-4dd48aec4a88.png)
4. Click user profile -> shares
5. Click on `urls/` on the left side
6. Send the following screen shot of what you got:
![image](https://user-images.githubusercontent.com/3269297/90749442-20d90f00-e2dc-11ea-925c-996f42cda486.png)
7. Also check the "preview" tab and let me know how many shares you have total.

Please also check this happens in the app as well as I think the app is already caching the results so the second load is fast.

**Comment by valleyofdawn** [comment]
![image](https://user-images.githubusercontent.com/689201/90751161-71ea0280-e2de-11ea-982c-18b214f9cbc4.png)

**Comment by valleyofdawn** [comment]
It happens in the app as well.

**Comment by HarelM** [comment]
Not exactly what I asked for, but to make it easier just send a screenshot of all the tabs.

**Comment by valleyofdawn** [comment]
The timing:
![image](https://user-images.githubusercontent.com/689201/90751689-1ec47f80-e2df-11ea-9a8a-39498911645c.png)

**Comment by valleyofdawn** [comment]
Any other tabs you need?

**Comment by HarelM** [comment]
Yes please - the "default" one - click the "x" in the corner to get to it, should look as follows:
![image](https://user-images.githubusercontent.com/3269297/90753135-e32ab500-e2e0-11ea-829d-fc0821529dc5.png)

**Comment by valleyofdawn** [comment]
![image](https://user-images.githubusercontent.com/689201/90753797-baef8600-e2e1-11ea-941b-8582ef35e813.png)

**Comment by HarelM** [comment]
Thanks!
Seems like there are two unrelated problem here:
1. The server is taking very long to return a very small amount of data - probably an issue with the indexing of the database.
2. The shares are not stored in the state for both (partial) offline usage and performance.

While 2 is very easy to solve and I'll probably implement it shortly, 1 is not.

Thanks for all the info! :-)

As a side note, you can compare it to the performance of the traces dialog, which has the caching code I'm planning for this dialog as well.
I'm guessing that solving 2 only will reduce the UX problem significantly.

**Comment by valleyofdawn** [comment]
The traces appear in 2.5 sec, and I have 200 of those (compares to 650 shares)

**Comment by HarelM** [comment]
@valleyofdawn Can you give it another test now?
